qualtran/bloqs/arithmetic/comparison.py

```diff
@@ -956,10 +957,62 @@ def wire_symbol(self, reg: Optional[Register], idx: Tuple[int, ...] = tuple()) -
             return TextBox(f"⨁(x = {self.val})")
         raise ValueError(f'Unknown register symbol {reg.name}')
 
+    def is_symbolic(self):
+        return is_symbolic(self.bitsize, self.val)
+
+    def build_composite_bloq(
+        self, bb: 'BloqBuilder', x: 'Soquet', target: 'Soquet'
+    ) -> Dict[str, 'SoquetT']:
```

Comment:
You can shorten the decomposition significantly by simply delegating to `MultiControlPauli(cvs=bits_k, target_gate=cirq.X).on(*x, *target)`. 


This is also cleaner IMO. We will soon update the `MultiControlPauli` bloq to not use `cirq.X` and instead use `XGate()` so that shouldn't be a blocker for using the bloq here.

Reply:
fixed